File: src/validparentheses/ValidParentheses.java
package validparentheses;


import java.util.Stack;

public class ValidParentheses {

    public static void main(String[] args) {
        String a = "(()){{([)]}}";
        boolean result = isValid(a);
        System.out.println("Result is " + result);
    }

    public static boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (c == '(')
                stack.push(')');
            else if (c == '{')
                stack.push('}');
            else if (c == '[')
                stack.push(']');
            else if (stack.isEmpty() || stack.pop() != c) {
                return false;
            }
        }
        return stack.isEmpty();
    }
}

File: src/Valid Parentheses/ValidParentheses.kt
package `Valid Parentheses`

import java.util.*

fun main() {

}

fun isValid(s: String): Boolean {
    val stack = Stack<Char>()
    s.reversed().forEach {
        when (it) {
            '(' -> if (stack.isNotEmpty() && stack.peek() == ')') stack.pop() else stack.push(it)
            '[' -> if (stack.isNotEmpty() && stack.peek() == ']') stack.pop() else stack.push(it)
            '{' -> if (stack.isNotEmpty() && stack.peek() == '}') stack.pop() else stack.push(it)
            else -> stack.push(it)
        }
    }

    return stack.isEmpty()
}